packages/core-data/src/queried-data/reducer.js

```diff
@@ -71,24 +81,30 @@ export function getMergedItemIds( itemIds, nextItemIds, page, perPage ) {
  *
  * @return {Object} Next state.
  */
-function items( state = {}, action ) {
+export function items( state = {}, action ) {
 	switch ( action.type ) {
-		case 'RECEIVE_ITEMS':
+		case 'RECEIVE_ITEMS': {
+			const context = getContextFromAction( action );
 			const key = action.key || DEFAULT_ENTITY_KEY;
 			return {
 				...state,
-				...action.items.reduce( ( accumulator, value ) => {
-					const itemId = value[ key ];
-					accumulator[ itemId ] = conservativeMapItem(
-						state[ itemId ],
-						value
-					);
-					return accumulator;
-				}, {} ),
+				[ context ]: {
+					...state[ context ],
+					...action.items.reduce( ( accumulator, value ) => {
+						const itemId = value[ key ];
+						accumulator[ itemId ] = conservativeMapItem(
+							state?.[ context ]?.[ itemId ],
+							value
+						);
+						return accumulator;
+					}, {} ),
+				},
 			};
+		}
 		case 'REMOVE_ITEMS':
-			const newState = omit( state, action.itemIds );
-			return newState;
+			return mapValues( state, ( contextState ) =>
+				omit( contextState, action.itemIds )
+			);
```

Comment:
Does this mean that query results for different contexts are stored separately, because the item data are actually different, while `removeItems` always removes the cached data for all contexts without discrimination?

Reply:
yes